radix-engine/src/system/node_modules/metadata/package.rs

```diff
@@ -53,7 +59,14 @@ impl MetadataNativePackage {
                     ));
                 }
 
-                Self::create(input, api)
+                let _input: MetadataCreateInput = scrypto_decode(&scrypto_encode(&input).unwrap())
+                    .map_err(|e| {
+                        RuntimeError::InterpreterError(InterpreterError::ScryptoInputDecodeError(e))
+                    })?;
+
+                let rtn = Self::create(api)?;
```

Comment:
I've been moving more of the decoding into the upper layer so that application logic doesn't get intermingled with process boundary logic. This is useful if we need to call these methods natively without crossing boundaries.